feat: sort migrations before applying

sf0rman opened the pull request:
# Problem Description

Migration files are not ordered by timestamp when running `up`-command which can cause ordering issues.

For example
Migration1: createCollection
Migration2: change stuff within collection

If not ordered, migration-stuff attempts to run first and fails due to dependency on Migration1 (or collection existing)
```
ExecuteMigrationError: MongoServerError: ns does not exist: db.collection
```

# Solution

When creating a migration file, we know that all class names end with a 13-digit timestamp (`./lib/commands/new.ts:57`) regardless of using a custom template or the default one.

We can therefore extract timestamp from className (last 13 digits of className) and sort migrations based on this before running migrations.

Commit:
feat: sort migrations before applying

matheusmelchiades reviewed (approved):
LGTM

matheusmelchiades commented:
Should check if order should be by file sort than className... 
https://github.com/mycodeself/mongo-migrate-ts/issues/114

mycodeself committed:
Merge branch 'master' into apply-migrations-in-order

mycodeself commented on lib/commands/up.ts:
```diff
@@ -40,12 +40,23 @@ export const up = async (opts: CommandUpOptions): Promise<void> => {
+            (m: MigrationModel) => m.className === migration.className
+          ) === undefined
+      )
+      .sort((a, b): number => {
+        // sort migrations by timestamp before applying
+        const aTimestamp = Number(
+          a.className.substring(a.className.length - 13)
+        );
+        const bTimestamp = Number(
+          b.className.substring(a.className.length - 13)
+        );
+        return aTimestamp > bTimestamp ? 1 : -1;
+      });
```
Can't we compare the className using `localeCompre` (compare as string) and not be dependent on the className format?

mycodeself replied:
```suggestion
      .sort((a, b) => a.className.localeCompare(b.className));
```

mycodeself commented on lib/utils/timestamp.ts:
```diff
@@ -3,13 +3,13 @@ import { format } from 'date-fns';
-      date.getUTCMinutes(),
-      date.getUTCSeconds(),
-      date.getUTCMilliseconds()
+      date.getFullYear(),
+      date.getMonth(),
+      date.getDate(),
+      date.getHours(),
+      date.getMinutes(),
+      date.getSeconds(),
+      date.getMilliseconds()
```
~~What motivates this change?~~

nvm... I see test are failing if not... 🤔

mycodeself reviewed (changes requested):
Hi @sf0rman, thank you very much for your input.

I have simply made a change to compare the className as a string with `localeCompare` when sorting the migrations, This way we are not dependent on the format of the className which may change.

I'll generate a fix version with these changes right now!

Thanks again!

Edit: I messed up it with the `localCompare` option, I have reverted to your original code in the version 1.6.2, it works perfectly. Thank you

mycodeself committed:
Update lib/commands/up.ts

Use localeCompare instead of relying on the format of the className that could change.

mycodeself commented:
:tada: This PR is included in version 1.6.1 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/mongo-migrate-ts/v/1.6.1)
- [GitHub release](https://github.com/mycodeself/mongo-migrate-ts/releases/tag/v1.6.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
